test/integration/additive/controls/additive.rb

```diff
@@ -45,14 +49,11 @@
 control 'folder-bindings' do
   title 'Test folder bindings are correct'
 
-  describe folder_bindings(folders[0]) do
-    it { should include role: folder_roles[0], members: member_groups[0] }
-    it { should include role: folder_roles[1], members: member_groups[1] }
-  end
-
-  describe folder_bindings(folders[1]) do
-    it { should include role: folder_roles[0], members: member_groups[0] }
-    it { should include role: folder_roles[1], members: member_groups[1] }
+  for folder in folders do
```

Comment:
Let's try to avoid dynamically defining test cases.

```ruby
describe "folder bindings" do
  it "include the first binding" do
    if roles < 1
      skip "less than 1 roles specified"
    else
      expect(folders).to all include role: folder_roles[0], members: member_groups[0]
    end
  end

  it "include the second binding" do
    if roles < 2
      skip "less than 2 roles specified"
    else
      expect(folders).to all include role: folder_roles[1], members: member_groups[1]
    end
  end
end
```

Reply:
Wow! That's an awesome way to restructure the tests, thanks! Changed all suites to use `all` and an explicit `skip`.